spec/strava/api/client/endpoints/uploads/create_upload_spec.rb

```diff
@@ -10,13 +10,34 @@
       file: Faraday::UploadIO.new(file, 'application/tcx+xml'),
       name: 'Uploaded Activity',
       description: 'Uploaded by strava-ruby-client.',
-      data_type: 'tcx',
-      external_id: 'strava-ruby-client-upload-1'
+      data_type: 'tcx'
     )
-    expect(upload.id).to eq 2_136_460_097
-    expect(upload.external_id).to eq 'strava-ruby-client-upload-1.tcx'
+
+    expect(upload.id).to eq 8_908_667_008
     expect(upload.error).to eq nil
     expect(upload.status).to eq 'Your activity is still being processed.'
     expect(upload.activity_id).to eq nil
   end
+  # rubocop:disable Style/MultilineBlockChain
+  it 'returns a Strava::Errors::UploadFailed on upload error', vcr: { cassette_name: 'client/create_upload_file_with_error' } do
+    expect do
+      # You need to wait when recording new cassettes, for the background job processing the upload to finish.
+      upload = client.create_upload(file: Faraday::UploadIO.new(file, 'application/tcx+xml'), data_type: 'tcx')
```

Comment:
Move this out of `expect`? That's not what throws an error I believe.

What does `client.upload` return during processing? Let's add a spec for that.

Reply:
> On a second thought, I think we need some way to tell when the upload finished. Is this when `upload.activity_id` is something? Add a `.completed?` method to that result so that the happy path can loop?

@dblock If params are missing our `Fault` error kicks in as http status isn't 200ish.  
If strava accepts the upload, an ID is returned. Which we can then use to query the status of the upload on Strava's side. That is why I added the comment about sleep in the spec I added.

I can put a little more work in and try to have the process mapped out much closer to what's going on IRL.